tests/conftest.py

```diff
@@ -1,90 +1,74 @@
-from flask.testing import FlaskClient
-import pytest
+import os
 
+import pytest
+from fastapi.testclient import TestClient
 
-def pytest_configure(config):
-    config.addinivalue_line("markers", "login_as")
+from airflow.api_fastapi.app import create_app
+from airflow.api_fastapi.auth.managers.simple.user import SimpleAuthManagerUser
+import datetime
+import time_machine
 
 
-@pytest.fixture
-def client(app, user, request):
-    """The test client, optionally logged in as a user of the given role
+API_PATHS = {
+    "public": "/api/v2",
+    "ui": "/ui",
+}
 
-    @pytest.mark.login_as('Admin')
-    def test_while_logged_in(client):
-        ....
+BASE_URL = "http://testserver"
 
-    def test_while_anon(client):
-        ...
 
-    """
-    marker = request.node.get_closest_marker('login_as')
-    user_obj = None
+def get_api_path(request):
+    """Determine the API path based on the test's subdirectory."""
+    test_dir = os.path.dirname(request.path)
+    subdirectory_name = test_dir.split("/")[-1]
 
-    if marker:
-        user_obj = user(marker.args[0])
+    return API_PATHS.get(subdirectory_name, "/")
 
-    with app.test_client(user=user_obj) as client:
-        yield client
 
-
-@pytest.fixture(scope='session')
-def app():
+@pytest.fixture
+def test_client(request, caplog):
     from airflow.utils.db import initdb
-    from airflow.www.app import create_app
 
-    initdb()
-    app = create_app(testing=True)
-    app.test_client_class = FlaskLoginClient
-    return app
+    os.environ["AIRFLOW__DATABASE__EXTERNAL_DB_MANAGERS"] = (
+        "astronomer.airflow.version_check.models.manager.VersionCheckDBManager"
+    )
 
+    # mimick password for simple auth manager
```

Comment:
It's probably easier to just use all_admins, then you don't need to worry about auth?

Reply:
Done. Thanks!